plugins/destination/managed_writer.go

```diff
@@ -0,0 +1,170 @@
+						p.logger.Err(err).Str("table", table.Name).Int("len", len(resources)).Dur("duration", time.Since(start)).Msg("failed to write last batch")
+						atomic.AddUint64(&metrics.Errors, uint64(len(resources)))
+					} else {
+						p.logger.Info().Str("table", table.Name).Int("len", len(resources)).Dur("duration", time.Since(start)).Msg("last batch written successfully")
+						atomic.AddUint64(&metrics.Writes, uint64(len(resources)))
+					}
+				}
+				return
+			}
+		case <-time.After(batchTimeout):
+			if len(resources) > 0 {
+				start := time.Now()
+				if err := p.client.WriteTableBatch(ctx, table, resources); err != nil {
+					p.logger.Err(err).Str("table", table.Name).Int("len", len(resources)).Dur("time", time.Since(start)).Msg("failed to write batch on timeout")
+					// we don't return as we need to continue until channel is closed otherwise there will be a deadlock
+					atomic.AddUint64(&metrics.Errors, uint64(len(resources)))
+				} else {
+					p.logger.Info().Str("table", table.Name).Int("len", len(resources)).Dur("time", time.Since(start)).Msg("batch written successfully on timeout")
+					atomic.AddUint64(&metrics.Writes, uint64(len(resources)))
+				}
+				resources = make([][]interface{}, 0)
+			}
+		case done := <-flush:
+			if len(resources) > 0 {
+				start := time.Now()
+				if err := p.client.WriteTableBatch(ctx, table, resources); err != nil {
+					p.logger.Err(err).Str("table", table.Name).Int("len", len(resources)).Dur("time", time.Since(start)).Msg("failed to write batch on timeout")
+					// we don't return as we need to continue until channel is closed otherwise there will be a deadlock
+					atomic.AddUint64(&metrics.Errors, uint64(len(resources)))
+				} else {
+					p.logger.Info().Str("table", table.Name).Int("len", len(resources)).Dur("time", time.Since(start)).Msg("batch written successfully on timeout")
+					atomic.AddUint64(&metrics.Writes, uint64(len(resources)))
+				}
+				resources = make([][]interface{}, 0)
+				done <- true
+			}
+		}
+	}
+}
+
+func (p *Plugin) writeManagedTableBatch(ctx context.Context, tables schema.Tables, sourceName string, syncTime time.Time, res <-chan schema.DestinationResource) error {
+	syncTime = syncTime.UTC()
+	SetDestinationManagedCqColumns(tables)
+
+	workers := make(map[string]*worker, len(tables))
+	metrics := &Metrics{}
+
+	p.workersLock.Lock()
+	// we call this after workers lock as we don't want to call PreWriteTableBatch concurrently
+	if err := p.client.PreWriteTableBatch(ctx, tables); err != nil {
+		p.workersLock.Unlock()
+		return err
+	}
+	for _, table := range tables.FlattenTables() {
+		table := table
+		if p.workers[table.Name] == nil {
+			ch := make(chan schema.CQTypes)
+			flush := make(chan chan bool)
+			wg := &sync.WaitGroup{}
+			p.workers[table.Name] = &worker{
+				count: 1,
+				ch:    ch,
+				flush: flush,
+				wg:    wg,
+			}
+			for i := 0; i < p.spec.Workers; i++ {
```

Comment:
Is the idea here to have multiple workers per table, or why are we ranging over the number of workers here? I'm a bit confused because the map above initializes count as 1. I may need to read through the code a bit more to understand

Reply:
Yeah, I removed this now there can be only one worker per table for now.